worker/backend/backend_test.go

```diff
@@ -260,240 +251,55 @@ func (s *BackendSuite) TestLookupGetContainer() {
-
-	fakeTask.WaitStub = func(ctx context.Context) (<-chan containerd.ExitStatus, error) {
-		c := make(chan containerd.ExitStatus, 1)
-		go func() {
-			es := containerd.NewExitStatus(0, time.Now(), errors.New("sigterm error"))
-			c <- *es
-			close(c)
-		}()
-		return c, nil
-	}
-
-	fakeContainer.TaskReturns(fakeTask, nil)
-	s.client.GetContainerReturns(fakeContainer, nil)
-
-	fakeTask.DeleteReturns(nil, nil)
-
-	err := s.backend.Destroy("some-handle")
-
-	s.Equal(1, fakeTask.KillCallCount())
-	s.EqualError(err, "client error: sigterm error")
-}
-
-func (s *BackendSuite) TestDestroyKillTaskTimeoutError() {
-	// so we don't have to wait 10 seconds for the default timeout
-	s.backend = backend.NewWithTimeout(s.client, testNamespace, 10*time.Millisecond)
-	fakeContainer := new(libcontainerdfakes.FakeContainer)
-	fakeTask := new(libcontainerdfakes.FakeTask)
-
-	fakeContainer.TaskReturns(fakeTask, nil)
-	exitChannel := make(chan containerd.ExitStatus) // this never returns
-	fakeTask.WaitReturns(exitChannel, nil)
-	s.client.GetContainerReturns(fakeContainer, nil)
-
-	fakeTask.KillReturnsOnCall(0, nil)
-	fakeTask.KillReturnsOnCall(1, errors.New("kill-again-error"))
-
-	err := s.backend.Destroy("some-handle")
-
-	s.Equal(2, fakeTask.KillCallCount())
-	_, firstSignal, _ := fakeTask.KillArgsForCall(0)
-	_, secondSignal, _ := fakeTask.KillArgsForCall(1)
-	s.Equal(firstSignal, syscall.SIGTERM)
-	s.Equal(secondSignal, syscall.SIGKILL)
-
-	s.EqualError(err, "client error: kill-again-error")
+	s.EqualError(err, "empty handle")
 }
 
-func (s *BackendSuite) TestDestroyDeleteTaskError() {
-	fakeContainer := new(libcontainerdfakes.FakeContainer)
-	fakeTask := new(libcontainerdfakes.FakeTask)
-
-	fakeTask.WaitStub = exitBeforeTimeout
-	fakeContainer.TaskReturns(fakeTask, nil)
-	s.client.GetContainerReturns(fakeContainer, nil)
-
-	fakeTask.KillReturns(nil)
-	fakeTask.DeleteReturns(nil, errors.New("task-delete-error"))
+func (s *BackendSuite) TestDestroyGetContainerError() {
+	s.client.GetContainerReturns(nil, errors.New("get-container-failed"))
 
 	err := s.backend.Destroy("some-handle")
-
-	s.Equal(1, fakeTask.KillCallCount())
-	s.Equal(1, fakeTask.DeleteCallCount())
-
-	ctx, _ := fakeTask.DeleteArgsForCall(0)
-	namespace, found := namespaces.Namespace(ctx)
-	s.True(found)
-	s.Equal(testNamespace, namespace)
-
-	s.EqualError(err, "client error: task-delete-error")
+	s.EqualError(errors.Unwrap(err), "get-container-failed")
 }
 
-func (s *BackendSuite) TestDestroyDeleteTaskFailedStatusError() {
-	fakeContainer := new(libcontainerdfakes.FakeContainer)
-	fakeTask := new(libcontainerdfakes.FakeTask)
-
-	fakeTask.WaitStub = exitBeforeTimeout
-	fakeContainer.TaskReturns(fakeTask, nil)
-	s.client.GetContainerReturns(fakeContainer, nil)
-
-	fakeTask.KillReturns(nil)
-	deleteStatus := containerd.NewExitStatus(1, time.Now(), errors.New("delete failed"))
-	fakeTask.DeleteReturns(deleteStatus, nil)
+// func (s *BackendSuite) TestDestroyGracefullyStopErrors() {
```

Comment:
Replace these tests to test the new Killer

Reply:
https://github.com/concourse/concourse/issues/5184